stdlib/public/runtime/CMakeLists.txt

```diff
@@ -52,20 +53,39 @@ add_swift_library(swiftRuntime IS_STDLIB IS_STDLIB_CORE
   C_COMPILE_FLAGS ${swift_runtime_compile_flags}
   INSTALL_IN_COMPONENT stdlib)
 
+set(object_target_list)
 foreach(sdk ${SWIFT_CONFIGURED_SDKS})
   if("${sdk}" STREQUAL "LINUX" OR "${sdk}" STREQUAL "FREEBSD")
     foreach(arch ${SWIFT_SDK_${sdk}_ARCHITECTURES})
       set(arch_subdir "${SWIFT_SDK_${sdk}_LIB_SUBDIR}/${arch}")
 
-      # FIXME: We will need a different linker script for 32-bit builds.
-      configure_file(
-          "swift.ld" "${SWIFTLIB_DIR}/${arch_subdir}/swift.ld" COPYONLY)
+      set(conformance_magic_begin_name "conformance_magic_begin_${SWIFT_SDK_${sdk}_ARCH_${arch}_TRIPLE}")
+      set(conformance_magic_end_name "conformance_magic_end_${SWIFT_SDK_${sdk}_ARCH_${arch}_TRIPLE}")
+      add_library(${conformance_magic_begin_name} STATIC swift_sections.cpp)
+      set_target_properties(${conformance_magic_begin_name} PROPERTIES COMPILE_FLAGS "-DSWIFT_BEGIN")
+      add_library(${conformance_magic_end_name} STATIC swift_sections.cpp)
+      set_target_properties(${conformance_magic_end_name} PROPERTIES COMPILE_FLAGS "-DSWIFT_END")
+
+      add_custom_command(
+        OUTPUT  "${SWIFTLIB_DIR}/${arch_subdir}/swift_begin.o"
+        COMMAND "${CMAKE_COMMAND}" -E copy "${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/${conformance_magic_begin_name}.dir/swift_sections.cpp${CMAKE_C_OUTPUT_EXTENSION}" "${SWIFTLIB_DIR}/${arch_subdir}/swift_begin.o"
```

Comment:
@gribozavr Unfortunately, no. Since the sources list specified in the add_library contains objects, we can't even use an OBJECT library. Cmake reorders objects to be put before anything else, including object library references, so the only way to have the right order is to use the objects generated from a previous target compilation. The correct fix is to avoid using objects at all, but that would require a lot of changes.